### PR TITLE
revive Ubuntu 20.04 packaging

### PR DESCRIPTION
**Pull request by christianparpart**
## Checklist

- [x] CI build workflow
- [x] CI release workflow to also publish .deb/.ddeb for release page
- [x] overall cleanup in github actions wrt packaging

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -595,6 +595,72 @@ jobs:
         with:
           name: contour-ubuntu2204-builddir
 
+  ubuntu_2004_packaging:
+    name: "Ubuntu 20.04: Create package"
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
```
Comment (data-man) [inline .github/workflows/build.yml:604]:
```suggestion
        uses: hendrikmuhs/ccache-action@v1.2.2
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -595,6 +595,72 @@ jobs:
...
+                "../contour-dbgsym-${{ steps.set_vars.outputs.VERSION_STRING }}-ubuntu_20_04_amd64.ddeb"
+      - name: "Uploading artifact .deb package"
+        uses: actions/upload-artifact@v2
+        with:
+          name: "contour-${{ steps.set_vars.outputs.VERSION_STRING }}-ubuntu_20_04_amd64.deb"
+          path: "contour-${{ steps.set_vars.outputs.VERSION_STRING }}-ubuntu_20_04_amd64.deb"
+          if-no-files-found: error
+          retention-days: 7
+      - name: "Uploading artifact .ddeb package (debugging symbols)"
+        uses: actions/upload-artifact@v2
```
Comment (data-man) [inline .github/workflows/build.yml:657]:
```suggestion
        uses: actions/upload-artifact@v3.1.0
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (christianparpart) [reply .github/workflows/build.yml:604]: Is this an in-place change that works out of the box or does it require more changes than simply adapting the number ?

Comment (data-man) [reply .github/workflows/build.yml:604]: Works out of the box I hope.
Also actions supports `@latest` tag if I'm not mistaken.

File: src/contour/CMakeLists.txt [hunk 3]
```diff
@@ -285,14 +285,17 @@ else()
...
+        set(DEPENDS_EXTRA ", libyaml-cpp0.5v5")
+        set(DEPENDS_EXTRA "${DEPENDS_EXTRA}, libboost-filesystem1.65.1")
+    elseif("${LSB_RELEASE_NUMBER}" STREQUAL "20.04")
+        set(DEPENDS_EXTRA ", libyaml-cpp0.6")
     endif()
 
     # Generator: Debian
     # see https://cmake.org/cmake/help/v3.6/module/CPackDeb.html#variable:CPACK_DEBIAN_PACKAGE_DEPENDS
-    set(CPACK_DEBIAN_PACKAGE_DEPENDS "libfreetype6, libharfbuzz0b, libqt5core5a, libqt5gui5, libqt5network5, ${YAML_CPP_PKG_NAME}")
+    set(CPACK_DEBIAN_PACKAGE_DEPENDS "libfreetype6, libharfbuzz0b, libqt5core5a, libqt5gui5, libqt5network5, libqt5multimedia5, libqt5x11extras5${DEPENDS_EXTRA}")
```
Comment (whisperity) [inline src/contour/CMakeLists.txt:298]: missing space / typo here?

Comment (christianparpart) [reply src/contour/CMakeLists.txt:298]: It's not missing. the `${DEPENDS_EXTRA}` is either an empty string or something like `", a"` or `", a, b"`

**Pull request merged**